src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -131,11 +133,15 @@ class CornerstoneViewport extends Component {
       viewport: cornerstone.getDefaultViewport(null, undefined)
     };
 
-    const { loadHandlerManager } = cornerstoneTools;
-    loadHandlerManager.setStartLoadHandler(this.startLoadingHandler);
-    loadHandlerManager.setEndLoadHandler(this.doneLoadingHandler);
-
     this.debouncedResize = debounce(() => {
+      loadHandlerManager.setStartLoadHandler(
```

Comment:
You have to move these calls too.

setStartLoadHandler should not be inside debouncedResize, it makes no sense at all.
Debounced resize is called every single time the viewport is resized. It happens 10s-100s of times.
The way you have the code now, every single time the viewport is resized, we will be pointlessly setting the handlers. There is no reason at all why this needs to be done. Move it out of this function.